### ISSUE TITLE
Image acquisition issues with Teledyne Dalsa Genie TS

### ISSUE THREAD

**Issue by TemporalArc** [SintefRaufossManufacturing/python-aravis#2]
I'm trying to use this library to acquire images from the genie ts. I've written code as per the examples, but my images are in grayscale and I get a lot of artifacts all over the images. Any idea what I could do to improve performance?

**Comment by oroulet** [comment]
all cameras are different, you need to find out what settings your camera want and set the attributes accodingly

**Comment by TemporalArc** [comment]
I suppose I should have mentioned this in the original comment, but I was trying to use the python interface. I have since implemented a c++ solution that works correctly with the same attributes I was using previously...so while my specific issue has been resolved, the devs might want look into the python interface.

Additional details:
My camera supports BayerGB8 and BayerGB10 pixel formats, however the frames being returned were 2d arrays instead of 3d. I believe this may also be the source of the artifacts, since under these settings the camera would be transmitting rgb values. The frames dimensions matched that of the expected image size, so it was not possible to reshape the matrix into a 3d array. Using Mono8, produced the expected results with smooth acquisition and no artifacts. All of this is when using the python interface.
